homeassistant/components/firmata/board.py

```diff
@@ -0,0 +1,185 @@
+        """Set up a Firmata instance."""
+        try:
+            _LOGGER.info("Connecting to Firmata %s", self.name)
+            self.api = await get_board(self.config)
+            self.firmware_version = await self.api.get_firmware_version()
+        except RuntimeError as err:
+            _LOGGER.error("Error connecting to PyMata board %s: %s", self.name, err)
+            return False
+
+        if CONF_SAMPLING_INTERVAL in self.config:
+            try:
+                await self.api.set_sampling_interval(
+                    self.config[CONF_SAMPLING_INTERVAL]
+                )
+            except RuntimeError as err:
+                _LOGGER.error(
+                    "Error setting sampling interval for PyMata \
+board %s: %s",
+                    self.name,
+                    err,
+                )
+                return False
+
+        if CONF_SWITCHES in self.config:
+            self.switches = self.config[CONF_SWITCHES]
+        if CONF_BINARY_SENSORS in self.config:
+            self.binary_sensors = self.config[CONF_BINARY_SENSORS]
+
+        _LOGGER.info("Firmata connection successful for %s", self.name)
+        return True
+
+    async def async_reset(self):
+        """Reset the board to default state."""
+        _LOGGER.debug("Shutting down board %s", self.name)
+        # If the board was never setup, continue.
+        if self.api is None:
+            return True
+
+        await self.api.shutdown()
+        self.api = None
+
+        return True
+
+    async def async_update_device_registry(self):
+        """Update board registry."""
+        device_registry = await self.hass.helpers.device_registry.async_get_registry()
+        device_registry.async_get_or_create(
+            config_entry_id=self.config_entry.entry_id,
+            connections={},
+            identifiers={(DOMAIN, self.name)},
+            manufacturer="Firmata",
+            name=self.name,
+            sw_version=self.firmware_version,
+        )
+
+
+async def get_board(data: dict):
+    """Create a Pymata board object."""
+    board_data = {}
+
+    if CONF_SERIAL_PORT in data:
+        board_data["com_port"] = data[CONF_SERIAL_PORT]
+    if CONF_SERIAL_BAUD_RATE in data:
+        board_data["baud_rate"] = data[CONF_SERIAL_BAUD_RATE]
+    if CONF_ARDUINO_INSTANCE_ID in data:
+        board_data["arduino_instance_id"] = data[CONF_ARDUINO_INSTANCE_ID]
+
+    if CONF_ARDUINO_WAIT in data:
+        board_data["arduino_wait"] = data[CONF_ARDUINO_WAIT]
+    if CONF_SLEEP_TUNE in data:
+        board_data["sleep_tune"] = data[CONF_SLEEP_TUNE]
+
+    board_data["autostart"] = False
+    board_data["shutdown_on_exception"] = True
+    board_data["close_loop_on_shutdown"] = False
+
+    board = PymataExpress(**board_data)
+
+    await board.start_aio()
+    return board
+
+
+class FirmataBoardPin(Entity):
+    """Manages a single Firmata board pin."""
+
+    def __init__(self, hass, config_entry, **kwargs):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Pass in the board instead.